app/models/application_record.rb

```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
```

Comment:
In addition, the `User`, `Post`, `Comment`, and `Like` models should have the following methods.

- User model should have a method that returns the 3 most recent posts.
`app/models/user.rb`
```
... //user associations code

def recent_post
  posts.order('created_at Desc').limit(3)
end
```

- Post model should have a method that updates the posts counter for a user.
`app/models/post.rb`
```
... //post associations code

def update_post_counter
  author.increment!(:posts_counter)
end
```

[OPTIONAL] I think it would be a great idea if you could also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.
```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Then you can also continue adding the following required methods:
- Kindly make sure that the Post model has a method that returns the `5 most recent comments` for a given post.
- Kindly ensure that the Comment model has a method that `updates the comments counter` for a post.
- Kindly ensure that the Like model has a method that `updates the likes counter` for a post.

[OPTIONAL] You can do the same thing for comments_counter and likes_counter by adding the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the methods every time a new comment or like has been saved to the database.